src/components/password/password.test.js

```diff
@@ -0,0 +1,178 @@
+      "should check fieldHelp renders properly with %s specific value",
+      (specificValue) => {
+        CypressMountWithProviders(
+          <PasswordComponent fieldHelp={specificValue} />
+        );
+
+        fieldHelpPreview().should("have.text", specificValue);
+      }
+    );
+
+    it.each(specialCharacters)(
+      "should check tooltip renders properly with %s specific values",
+      (specificValue) => {
+        CypressMountWithProviders(
+          <PasswordComponent labelHelp={specificValue} />
+        );
+
+        getDataElementByValue("question").trigger("mouseover");
+        tooltipPreview().should("have.text", specificValue);
+      }
+    );
+
+    it("should check add icon inside of the Password component renders", () => {
+      CypressMountWithProviders(<PasswordComponent inputIcon="add" />);
+
+      getDataElementByValue("add").should("be.visible");
+    });
+
+    it("should check the Password component is disabled", () => {
+      CypressMountWithProviders(<PasswordComponent disabled />);
+
+      commonDataElementInputPreview().parent().should("have.attr", "disabled");
+      commonDataElementInputPreview().should("be.disabled");
+    });
+
+    it("should check the Password component is required", () => {
+      CypressMountWithProviders(<PasswordComponent required />);
+
+      verifyRequiredAsteriskForLabel();
+    });
+
+    it("should check the Password component has autofocus", () => {
+      CypressMountWithProviders(<PasswordComponent autoFocus />);
+
+      commonDataElementInputPreview().should("be.focused");
+    });
+
+    it.each([
+      ["right", "end"],
+      ["left", "start"],
+    ])(
+      "should use %s as labelAligment and render it with %s as css properties",
+      (alignment, cssProp) => {
+        CypressMountWithProviders(
+          <PasswordComponent labelInline labelAlign={alignment} />
+        );
+
+        getDataElementByValue("label")
+          .parent()
+          .should("have.css", "-webkit-box-pack", cssProp)
+          .and("have.css", "justify-content", `flex-${cssProp}`);
+      }
+    );
+
+    it.each(["10%", "30%", "50%", "80%", "100%"])(
+      "should check maxWidth as %s for Password component",
+      (maxWidth) => {
+        CypressMountWithProviders(<PasswordComponent maxWidth={maxWidth} />);
+
+        getDataElementByValue("input")
+          .parent()
+          .parent()
+          .should("have.css", "max-width", maxWidth);
+      }
+    );
+
+    it("when maxWidth has no value it should render as 100%", () => {
+      CypressMountWithProviders(<PasswordComponent maxWidth="" />);
+
+      getDataElementByValue("input")
+        .parent()
+        .parent()
+        .should("have.css", "max-width", "100%");
+    });
+  });
+
+  describe("check events for Password component", () => {
+    let callback;
+
+    beforeEach(() => {
+      callback = cy.stub();
+    });
+
+    it("should call onBlur callback when a blur event is triggered", () => {
```

Comment:
don't we have `onChange`/`onClick`/`onFocus` events here?

Reply:
Yeah good point, in retrospect I think we can remove any event tests as they're all covered in `Textbox`, which this component renders. Happy to include them though if needed 👍

Reply:
Yeah, I don't think you need the event tests because, as you say, they're covered in Textbox. If you were hijacking these events to do something specific to this component, it would be needed then but I think you're fine to not include these.